dev/test/traverse.test.ts

```diff
@@ -95,7 +97,30 @@ describe("traverse", () => {
         attest(t({ a: 5 }).problems?.summary).snap(
             "a must be a string or boolean (was 5)"
         )
+        attest(t.flat).snap([
+            ["domain", "object"],
+            [
+                "switch",
+                {
+                    path: ["a"],
+                    kind: "domain",
+                    cases: { string: [], boolean: [] }
+                }
+            ]
+        ])
+    })
+    it("multiple switch", () => {
+        const $ = scope({
```

Comment:
`$` is usually for a scope.

This is a Space since compile was called, so it's usually called `types`